### PR TITLE
[#54] [Chore] Add Dependency Injection framework: Factory

### LINKED ISSUE #54: [Chore] Add Dependency Injection framework: Factory
## Why

Currently, we are not using any frameworks or libraries to apply the `Inversion of Control` design pattern. We need to support it for creating ViewModels, Usecases, Repositories, etc. by using [Factory](https://github.com/hmlongco/Factory) framework (the successor of the [Resolver](https://github.com/hmlongco/Resolver) framework).

## Acceptance Criteria

- Add [Factory library](https://github.com/hmlongco/Factory) into the project's SPM.
- Set up some setup classes and foundation injections for later usage.
- Convert existing initializations to use the new DI library.
 
## Who Benefits?

Developers need to access and manage the ViewModels, Usecases, and Repositories instances while developing the app and feature modules.

### PR DESCRIPTION
**Pull request by minhnimble**
- Close #54

## What happened 👀

- Add [Factory library](https://github.com/hmlongco/Factory) into the project's SPM.
- Set up some setup classes and foundation injections for later usage via the `Container+Injection.swift` file.
- Convert existing initialization to use the new DI library in `CryptoPricesApp` e.g. `HomeViewModel` injection.

## Insight 📝

During the implementation, I encountered a swiftlint issue on the `HomeViewModelSpect.swift` that is discussed [here](https://nimble-co.slack.com/archives/C04C7E5PWUQ/p1671596381472819). In the end, the fix for now is to update the rules to `// swiftlint:disable function_body_length closure_body_length superfluous_disable_command`. We might need to tackle the root cause to make sure the linting on the CI environment should be the same with the local at a later time. 🙏

I also added grouped folders in the main project for better organization:
- `Dependency Injection`folder: contains  `Container+Injection.swift` file.
- `Navigation` folder: contains `AppCoordinator.swift` file and `AppState.swift` file.

Since our setup for the Navigation is on the `CryptoPricesApp`, there is no proper way to setup the Dependencies Injection on each feature modules that there will be plenty of duplicated library imports and DI setup codes, Therefore, I decide to apply DI at the same place with the current Navigation setup code, which is in the main application. 💪

## Proof Of Work 📹

No visible changes, just under the hood optimization and the application built and ran successfully. 🛠🎉

Screenshot 2022-12-22 at 09 53 13

### CODE REVIEW

File: CryptoPrices/Sources/App/Dependencie Injection/Container+Injection.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import Home
+import MyCoin
+import NetworkExtension
+import UseCaseProtocol
+import NetworkCore
+import Pilot
+import Repositories
+import RepositoryProtocol
+import Styleguide
+import SwiftUI
```
Comment (Thieurom) [inline CryptoPrices/Sources/App/Dependencie Injection/Container+Injection.swift:18]: Just a minor thing, I believe we can omit these 2 imports `Styleguide` and `SwiftUI` here 🙏

**Review by Thieurom — APPROVED** [review]
Really nice work 👍

**Review by nkhanh44 — APPROVED** [review]
Great work 🚀

File: CryptoPrices/Sources/App/Dependencie Injection/Container+Injection.swift (hunk 1 above)
Comment (minhnimble) [reply CryptoPrices/Sources/App/Dependencie Injection/Container+Injection.swift]: you are right, nice catch! Fixed on [5aaaed2](https://github.com/nimblehq/nimble-crypto-ios/pull/65/commits/5aaaed210a7d1f7c1fbb6c1f0ecc2ef054c5dba3)

**Pull request merged**